### DISCUSSION TITLE
How to style a custom HTML attribute

### DISCUSSION THREAD

**Discussion by piotrkulpinski** [tailwindlabs/tailwindcss#12840] [Help]
Hi, I'm using the [Vaul](https://vaul.emilkowal.ski/) package which adds custom HTML attributes for drawer elements like `vaul-drawer-direction="right"`. 

Is there a way to style that with inline classes without modifying TailwindCSS config? 

Tried something like this but it doesn't work:
```
[vaul-drawer-direction=right]:right-0
```

Thanks

**Comment by wongjn** [comment]
For CSS attributes as arbitrary variants, the correct syntax would be something like `[&[attribute=value]]:` if the class would be on the same element as the `attribute`. So for your example, it could be something like:

```jsx
<Drawer.Content className="[&[vaul-drawer-direction=right]]:right-0">
```

**Reply by piotrkulpinski** [reply]
Exactly what I needed! Thanks 🙏
